File: back-end/domains/user/request.go
package user

import (
	"mime/multipart"
	"time"

	"github.com/devanadindraa/NTTH-Store/back-end/utils/constants"
)

type LoginReq struct {
	Email    string         `json:"email" valdiate:"required"`
	Password string         `json:"password" validate:"required"`
	Role     constants.ROLE `json:"role" validate:"omitempty,oneof=ADMIN CUSTOMER"`
}

type LogoutReq struct {
	Token   string
	Expires time.Time
}

type RegisterReq struct {
	Name     string `json:"name" validate:"required"`
	Password string `json:"password" validate:"required"`
	Email    string `json:"email" validate:"required,email"`
}

type ChangePasswordReq struct {
	CurrentPassword string `json:"current_password" validate:"required"`
	NewPassword     string `json:"new_password" validate:"required"`
}

type UpdateProfileReq struct {
	Name             string `json:"name" validate:"required"`
	Email            string
	DestinationID    string
	Address          string
	ZipCode          string
	DestinationLabel string
}

type AvatarReq struct {
	AvatarUrl *multipart.FileHeader `json:"AvatarUrl" validate:"required"`
}

type ResetPasswordReq struct {
	Email string
	Role  constants.ROLE `json:"role" validate:"omitempty,oneof=ADMIN CUSTOMER"`
}

type ResetPasswordSubmitReq struct {
	Email       string
	NewPassword string
	Role        constants.ROLE `json:"role" validate:"omitempty,oneof=ADMIN CUSTOMER"`
}

File: mobile/app/src/main/java/com/example/signlink/data/models/customer/Request.kt
package com.example.signlink.data.models.customer

import com.google.gson.annotations.SerializedName
import java.io.File

data class RegisterReq(
    val name: String,
    val email: String,
    val password: String
)

data class LoginReq(
    val role: String,
    val email: String,
    val password: String,
)

data class ResetPasswordReq(
    val email: String,
    val role: String,
)
data class ResetPasswordSubmit(
    val email: String,
    val newPassword: String,
    val role: String,
)

data class UpdateProfileReq(
    val name: String,
    val email: String
)

data class ChangePaswordReq(
    @SerializedName("current_password")
    val currentPassword: String,

    @SerializedName("new_password")
    val newPassword: String
)

data class AvatarReq(
    val avatar: File
)